connection.py

```
@@ -108,137 +121,264 @@ def log_to_csv(self, sample_data):
+                                next_sample_time = current_time + SAMPLE_INTERVAL
+                            
+                            if self.lsl_connection:
+                                self.lsl_connection.push_sample(channel_data, timestamp=sample_time)
+
+                            self.update_sample_rate()
+
+                            if self.recording_active:
+                                self.log_to_csv(channel_data)
+            except Exception as e:
+                print(f"USB data handler error: {str(e)}")
+                break
```

Pylint (3.3.7)

[refactor] Too many nested blocks (6/5)

(R1702)